### PR TITLE
fix(plasma-tokens): Fix gradient generator for iOS platform

### PR DESCRIPTION
**Pull request by neretin-trike**
:baby_chick: Download canary assets:

[colors--canary.211.3425447069.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.211.3425447069.xml)  
[color_sbermarket_ios-swift--canary.211.3425447069.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.211.3425447069.swift)  
[color_sbermarket_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.211.3425447069.kt)  
[color_sbermarket_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.211.3425447069.ts)  
[color_sberprime_ios-swift--canary.211.3425447069.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.211.3425447069.swift)  
[color_sberprime_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.211.3425447069.kt)  
[color_sberprime_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.211.3425447069.ts)  
[PlasmaTokensColor--canary.211.3425447069.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.211.3425447069.swift)  
[typo_mage_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.211.3425447069.kt)  
[typo_mage_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.211.3425447069.ts)  
[typo_plasma_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.211.3425447069.kt)  
[typo_plasma_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.211.3425447069.ts)  
[typo_ruler_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.211.3425447069.kt)  
[typo_ruler_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.211.3425447069.ts)  
[typo_sage_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.211.3425447069.kt)  
[typo_sage_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.211.3425447069.ts)  
[typo_sbermarket_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.211.3425447069.kt)  
[typo_sbermarket_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.211.3425447069.ts)  
[typo_soulmate_kotlin--canary.211.3425447069.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.211.3425447069.kt)  
[typo_soulmate_react-native--canary.211.3425447069.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.211.3425447069.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.111.2-canary.211.3425447069.0
  npm install @salutejs/plasma-core@1.81.2-canary.211.3425447069.0
  npm install @salutejs/plasma-icons@1.106.2-canary.211.3425447069.0
  npm install @salutejs/plasma-temple@1.109.4-canary.211.3425447069.0
  npm install @salutejs/plasma-tokens@1.28.2-canary.211.3425447069.0
  npm install @salutejs/plasma-ui@1.143.4-canary.211.3425447069.0
  npm install @salutejs/plasma-web@1.140.2-canary.211.3425447069.0
  npm install @salutejs/plasma-cy-utils@0.34.2-canary.211.3425447069.0
  npm install @salutejs/plasma-sb-utils@0.79.2-canary.211.3425447069.0
  npm install @salutejs/plasma-tokens-android@2.35.2-canary.211.3425447069.0
  npm install @salutejs/plasma-tokens-ios-swift@2.35.2-canary.211.3425447069.0
  # or 
  yarn add @salutejs/plasma-b2c@1.111.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-core@1.81.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-icons@1.106.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-temple@1.109.4-canary.211.3425447069.0
  yarn add @salutejs/plasma-tokens@1.28.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-ui@1.143.4-canary.211.3425447069.0
  yarn add @salutejs/plasma-web@1.140.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-cy-utils@0.34.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-sb-utils@0.79.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-tokens-android@2.35.2-canary.211.3425447069.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.35.2-canary.211.3425447069.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-211/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-211/
  ui storybook: http://plasma.sberdevices.ru/pr-211/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-211/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-211/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-211/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-211/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-211/
  ui storybook: http://plasma.sberdevices.ru/pr-211/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-211/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-211/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-211/temple-storybook/

File: packages/plasma-tokens/lib/generator/formatters/colorIosSwift.ts [hunk 1]
```diff
@@ -23,45 +25,130 @@ extension UIColor {
...
+                )
+            ],
+            startPoint: ${startPoint},
+            endPoint: ${endPoint}
+        )`;
+
 export const colorIosSwiftCustomFormatter = ({ dictionary, file }: { dictionary: Dictionary; file: File }) => {
     const themeName = file.className || '';
-    const filterWord = 'light';
+    const lightWord = 'light';
```
Comment (Yakutoc) [inline packages/plasma-tokens/lib/generator/formatters/colorIosSwift.ts:95]: ИМХО, твоя идея с вынести в **enum** мне кажется более приемлемой.

File: packages/plasma-tokens/lib/generator/formatters/colorIosSwift.ts [hunk 2]
```diff
@@ -23,45 +25,130 @@ extension UIColor {
...
+    const darkWord = 'dark';
+    const gradientWord = 'gradient';
 
     const struct = dictionary.allTokens
-        .filter((tokens) => tokens.attributes?.type?.includes(filterWord))
+        .filter(
+            (tokens) =>
```
Comment (Yakutoc) [inline packages/plasma-tokens/lib/generator/formatters/colorIosSwift.ts:101]: [Minor]

Сделать деструктуризацию, типо 

`filter(({ attributes, name }) => ...)`

File: packages/plasma-tokens/lib/generator/formatters/colorIosSwift.ts [hunk 3]
```diff
@@ -23,45 +25,130 @@ extension UIColor {
...
                 name,
-            )} = ColorToken(plasmaTokensColorLight: PlasmaTokensColor.light${name}, plasmaTokensColorLDark: PlasmaTokensColor.dark${name})`;
+            )} = ColorToken(plasmaTokensColorLight: PlasmaTokensColor.light${name}, plasmaTokensColorDark: PlasmaTokensColor.dark${name})`;
+        })
+        .join(`\n`);
+
+    const gradient = dictionary.allTokens
+        .filter((tokens) => tokens.name.toLocaleLowerCase().includes(gradientWord))
+        .reduce<Record<string, Record<ThemeColor, GradientToken>>>((acc, token) => {
+            const regex = new RegExp(`((${lightWord}|${darkWord})${themeName})`);
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/generator/formatters/colorIosSwift.ts:117]: ну кстати нет, спасиб 👍

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-211/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-211/
  ui storybook: http://plasma.sberdevices.ru/pr-211/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-211/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-211/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-211/temple-storybook/

**Pull request merged**